packages/vite/src/node/optimizer/scan.ts

```diff
@@ -434,9 +447,13 @@ function esbuildScanPlugin(
         {
           filter: /.*/
         },
-        async ({ path: id, importer }) => {
+        async ({ path: id, importer, pluginData }) => {
           // use vite resolver to support urls and omitted extensions
-          const resolved = await resolve(id, importer)
+          const resolved = await resolve(id, importer, {
+            custom: {
+              depScan: { loader: pluginData?.htmlType?.loader }
```

Comment:
We already have a `scan` flag that is available also to the resolver. But I agree that using the convention for `custom` here feels better. So looks good to me 👍🏼